core/modules/system/system.module

```diff
@@ -3903,6 +3903,27 @@ function system_403() {
 }
 
 /**
+ * Menu callback; Redirects a deprecated menu location to a new path.
+ *
+ * This wrapper around backdrop_goto_deprecated() should be used as the menu
+ * callback in hook_menu() when redirecting from an old path to a new one.
+ *
+ * @since 1.8.0
+ */
+function system_redirect_deprecated_page($path) {
+  // Perform any substitutions in the passed-in path with the current path.
+  $parts = explode('/', $path);
+  foreach ($parts as $index => $path_part) {
+    if (substr($path_part, 0, 1) === '%') {
+      $parts[$index] = arg($index);
+    }
```

Comment:
This assumes the parameters are in the same order and in the same place in both the old and new paths.  My first thought is to try to auto-map old to new if they're named (e.g., %node or %user), but that won't work with most paths.  What do you think about adding a second parameter to the function that maps the old to the new?  With that change, the `page arguments` would look something like this:

```php
  $menu['path/to/old/%/%/page/%'] = array(
    'page callback' => 'system_redirect_deprecated_page', 
    'page arguments' => array('path/%/to/%/new/page', array(1 => 4, 3 => 3)),
  );
```

If the second argument is missing, then a 1:1 mapping can be assumed.

Reply:
Sure, that's not difficult to add and might be useful. It doesn't make it more difficult for existing 1:1 maps so seems like a general win.